errors.go

```diff
@@ -122,9 +122,21 @@ func (e NotAcceptableError) StatusCode() int { return http.StatusNotAcceptable }
 func (e NotAcceptableError) Unwrap() error { return HTTPError(e) }
 
 // ErrorHandler is the default error handler used by the framework.
-// It transforms any error into the unified error type [HTTPError],
-// Using the [ErrorWithStatus] interface.
+// If the error is an [HTTPError] that is error is returned.
+// If the error adheres to the [ErrorWithStatus] interface
+// the error is transformed to a [HTTPError].
+// If the error is not an [HTTPError] nor does it adhere to an
+// interface the error is returned.
 func ErrorHandler(err error) error {
+	var errorStatus ErrorWithStatus
+	if errors.As(err, &HTTPError{}) || errors.As(err, &errorStatus) {
```

Comment:
You are not using error status

So maybe you could use the same syntax for both error types


```suggestion
	if errors.As(err, &HTTPError{}) || errors.As(err, &ErrorWithStatus{}) {
```

Also, ypu could increase readability by using a switch


```suggestion
	switch {
		case errors.As(err, &HTTPError{}),
			errors.As(err, &ErrorWithStatus{}):
```

Reply:
`ErrorWithStatus` is an interface. The code above isn't valid. 

<img width="777" alt="image" src="https://github.com/user-attachments/assets/877d955e-0f8b-4ddc-9b54-339c46272d2e">